src/psyclone/psyad/adjoint_visitor.py

```diff
@@ -361,6 +367,59 @@ def loop_node(self, node):
         new_node.children[3] = self._visit(node.children[3])
         return new_node
 
+    def ifblock_node(self, node):
+        '''This method is called if the visitor finds an ifblock node. An
+        exception is raised if the condition of the ifblock node
+        contains an active variable as this is not valid
+        tangent-linear code. Otherwise, the ifblock and its condition
+        are returned unchanged and the contents of the "then" and
+        "else" parts of the ifblock are returned after being processed
+        by PSyAD.
+
+        :param node: an IfBlock PSyIR node.
+        :type node: :py:class:`psyclone.psyir.nodes.IfBlock`
+
+        :returns: a new PSyIR tree containing the adjoint equivalent \
+            of this node and its descendants.
+        :rtype: :py:class:`psyclone.psyir.nodes.IfBlock`
+
+        :raises: VisitorError if the condition of the ifblock contains \
+            any active variables.
+        :raises VisitorError: if the ifblock node is visited before a \
+            schedule.
+        :raises VisitorError: if a passive ifblock node is found.
+
+        '''
+        if self._active_variables is None:
+            raise VisitorError(
+                "An ifblock node should not be visited before a schedule, "
+                "as the latter sets up the active variables.")
+
+        if node_is_active(node.condition, self._active_variables):
+            raise VisitorError(
+                f"The if condition '{self._writer(node.condition)}' of an "
+                f"ifblock node should not contain an active variable.")
```

Comment:
I suggest adding the list of active variable names to this error message.

Reply:
Added.